### PR TITLE
test: add allure public reports

### PR DESCRIPTION
**Pull request by pcheremu**
# What ❔

updated workflow to make public allure reports

## Why ❔

current reports provide a link to a private url

## Checklist

- [ + ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ n/a ] Tests for the changes have been added / updated.
- [ n/a ] Documentation comments have been added / updated.

### CODE REVIEW

File: .github/workflows/app-deploy-feature-branch.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ jobs:
     uses: ./.github/workflows/app-e2e.yml
```
[the remark this answers is not in the document]
Comment (pcheremu) [reply .github/workflows/app-deploy-feature-branch.yml:70]: @abilevych Sorry, I looked for this comment but I did not see them (presumably due to rebase). Sure, It make sense to add this description to the Readme.md. I will add.

File: .github/workflows/app-e2e.yml [hunk 2]
```diff
@@ -110,27 +116,21 @@ jobs:
...
-
-      - name: Upload tests to the Allure proj
-        if: always() && inputs.publish_to_allure == true
         env:
           ALLURE_TOKEN: ${{ secrets.ALLURE_TOKEN }}
         run: |
-          ./allurectl upload allure-results --launch-id ${{ env.ALLURE_LAUNCH_ID }}
-          ./allurectl launch close ${{ env.ALLURE_LAUNCH_ID }}
+          ./allurectl upload allure-results
+          echo "*Public report link: https://raw.githack.com/matter-labs/block-explorer/gh-pages/${{ github.run_number }}/index.html"
```
[the remark this answers is not in the document]
Comment (pcheremu) [reply .github/workflows/app-e2e.yml:132]: @abilevych To be honest I faced with such behavior when I clicked this link before files was cached by raw.githack.com. I had to wait some time in this case.

File: .github/workflows/app-deploy-feature-branch.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pcheremu) [reply .github/workflows/app-deploy-feature-branch.yml:70]: Added

File: .github/workflows/app-e2e.yml [hunk 3]
```diff
@@ -110,27 +116,21 @@ jobs:
...
         if: always()
-        run: |
-          echo "MATRIX_TAG_WITHOUT_QUOTES=$(echo ${{ matrix.tags }} | sed -e 's/@//g' )" >> $GITHUB_ENV
+        uses: actions/upload-artifact@v3
+        with:
+          name: allure-results
+          path: packages/app/allure-results
 
-      - name: Create launch ID
+      - name: Upload tests to Allure
```
[the remark this answers is not in the document]
Comment (pcheremu) [reply .github/workflows/app-e2e.yml]: done

File: .github/workflows/app-e2e.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pcheremu) [reply .github/workflows/app-e2e.yml:132]: Cannot be reproduced with a default use case.

**Review by vasyl-ivanchuk — APPROVED** [review]

**Pull request merged**